Action and workflow fail on latest runner: "you have held broken packages"

### Describe the problem

The reusable workflow runs in the `ubuntu-latest` [runner machine](https://docs.github.com/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners):

https://github.com/arduino/render-docs-github-action/blob/ed6472d4c2f6de94acc7c451d4fd0d60fa44b881/.github/workflows/render-docs.yml#L48

End user workflows are most likely to use the action in a job with either that convenient `ubuntu-latest` runner, or else the latest version specific `ubuntu-24.04`.

GitHub recently [changed the `ubuntu-latest` runner label to point to the `ubuntu-24.04` runner](https://github.com/actions/runner-images/issues/10636) instead of the `ubuntu-22.04` runner.

🐛 The action and reusable workflow fail on the `ubuntu-24.04` runner machine:

```text
  # Import LLVM GPG key without using apt-key
  wget -q -O - https://apt.llvm.org/llvm-snapshot.gpg.key | gpg --dearmor -o /usr/share/keyrings/llvm-archive-keyring.gpg

  # Add the LLVM repository with the GPG keyring
  echo "deb [signed-by=/usr/share/keyrings/llvm-archive-keyring.gpg] http://apt.llvm.org/focal/ llvm-toolchain-focal-9 main" | sudo tee /etc/apt/sources.list.d/llvm.list > /dev/null

  # Update the package lists
  sudo apt-get -qq update

  # Install libclang and other necessary packages
  sudo apt-get -qq install -y libclang1-9 libclang-cpp9 > /dev/null
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
E: Unable to correct problems, you have held broken packages.
Error: Process completed with exit code 100.
```

### To reproduce

#### Reusable workflow

Use the reusable workflow in any workflow.

```yaml
on:
  push:
permissions:
  contents: write
jobs:
  render-docs:
    uses: arduino/render-docs-github-action/.github/workflows/render-docs.yml@main
    with:
      source-path: './src'
      target-path: './docs/api.md'
```

#### Action

Use the action in a workflow job that uses the `ubuntu-latest` or `ubuntu-24.04` runner:

```yaml
on:
  push:
permissions:
  contents: write
jobs:
  render-docs:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v4
    - uses: arduino/render-docs-github-action@main
      with:
        source-path: './src'
        target-path: './docs/api.md'
```

---

🐛 The workflow run fails with an "`E: Unable to correct problems, you have held broken packages.`" error.

### Expected behavior

Action can be used with the latest GitHub Hosted Ubuntu runner.

Reusable workflow is functional.

### arduino/render-docs-github-action version

ed6472d4c2f6de94acc7c451d4fd0d60fa44b881

### Additional context

The error does not occur if the `ubuntu-22.04` runner is used.

#### Related

- https://github.com/actions/runner-images/issues/9546

<a name="workaround"></a>

#### Workaround

When using the action, the fault can be worked around by configuring the workflow job to use the `ubuntu-22.04` runner:

```yaml
runs-on: ubuntu-22.04
```

--- a/.github/workflows/render-docs.yml
+++ b/.github/workflows/render-docs.yml
@@ -45,7 +45,7 @@ on:
 
 jobs:
   render-docs:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
     permissions:
       contents: write
     steps:
@@ -64,13 +64,13 @@ jobs:
         wget -q -O - https://apt.llvm.org/llvm-snapshot.gpg.key | gpg --dearmor -o /usr/share/keyrings/llvm-archive-keyring.gpg
 
         # Add the LLVM repository with the GPG keyring
-        echo "deb [signed-by=/usr/share/keyrings/llvm-archive-keyring.gpg] http://apt.llvm.org/focal/ llvm-toolchain-focal-9 main" | sudo tee /etc/apt/sources.list.d/llvm.list > /dev/null
+        echo "deb [signed-by=/usr/share/keyrings/llvm-archive-keyring.gpg] http://apt.llvm.org/jammy/ llvm-toolchain-jammy-15 main" | sudo tee /etc/apt/sources.list.d/llvm.list > /dev/null
         
         # Update the package lists
         sudo apt-get -qq update
 
         # Install libclang and other necessary packages
-        sudo apt-get -qq install -y libclang1-9 libclang-cpp9 > /dev/null
+        sudo apt-get install -y libclang1-15 libclang-cpp15
 
     - name: Set up Node.js
       uses: actions/setup-node@v4
@@ -79,7 +79,7 @@ jobs:
         
     - name: Install render-docs
       shell: bash
-      run: npm install github:arduino/render-docs --no-save --silent
+      run: npm install github:arduino/render-docs --no-save
 
     - name: Cache Doxygen binaries
       id: cache-doxygen-binaries

--- a/action.yml
+++ b/action.yml
@@ -42,13 +42,13 @@ runs:
       wget -q -O - https://apt.llvm.org/llvm-snapshot.gpg.key | gpg --dearmor -o /usr/share/keyrings/llvm-archive-keyring.gpg
 
       # Add the LLVM repository with the GPG keyring
-      echo "deb [signed-by=/usr/share/keyrings/llvm-archive-keyring.gpg] http://apt.llvm.org/focal/ llvm-toolchain-focal-9 main" | sudo tee /etc/apt/sources.list.d/llvm.list > /dev/null
+      echo "deb [signed-by=/usr/share/keyrings/llvm-archive-keyring.gpg] http://apt.llvm.org/jammy/ llvm-toolchain-jammy-15 main" | sudo tee /etc/apt/sources.list.d/llvm.list > /dev/null
       
       # Update the package lists
       sudo apt-get -qq update
 
       # Install libclang and other necessary packages
-      sudo apt-get -qq install -y libclang1-9 libclang-cpp9 > /dev/null
+      sudo apt-get install -y libclang1-15 libclang-cpp15
 
   - name: Set up Node.js
     uses: actions/setup-node@v4
@@ -57,7 +57,7 @@ runs:
       
   - name: Install render-docs
     shell: bash
-    run: npm install github:arduino/render-docs --no-save --silent
+    run: npm install github:arduino/render-docs --no-save
 
   - name: Cache Doxygen binaries
     id: cache-doxygen-binaries

